ldap-api/src/main/java/ca/bc/gov/hlth/ldapapi/service/LdapService.java

```diff
@@ -83,6 +87,36 @@ private boolean checkUserLocked(Attributes attributes) {
         return unlocked;
     }
 
+    private boolean checkUserPasswordExpired(Attributes attributes) {
+        Attribute passwordChangeDate = attributes.get(LDAP_ATTR_PASSWORD_CHANGE_DATE);
+        Attribute passwordLifespan = attributes.get(LDAP_ATTR_PASSLIFESPAN);
+        boolean expired = false;
+
+        if (passwordChangeDate != null) {
+            try {
+                Long changedDaysAgo = Long.valueOf(passwordChangeDate.get().toString());
+                Long today = LocalDate.now().toEpochDay();
+                expired = (changedDaysAgo + checkPasswordLifespan(passwordLifespan)) < today;
+            } catch (NamingException e) {
+                e.printStackTrace();
```

Comment:
You have a logger defined in this class, say you may as well log the stacktrace instead of printing it to System.out.

Why would a `NamingException` be thrown? It looks like it throws a `NoSuchElementException` if the attribute has no value. 

Personally I'd feel safer if the code just re-threw the exception as a RuntimeException. We should [fail securely](https://owasp.org/www-community/Fail_securely). If you like that approach, you can use a method like this:

```java
private Object getAttributeValue(Attribute att) {
    try {
        return att.get();
    } catch (NamingException e) {
        throw new RuntimeException(e);
    }
}
```